### DISCUSSION TITLE
Einbindung externer Sensoren in HmIP

### DISCUSSION THREAD

**Discussion by TheBoringGuy1908** [OpenCCU/OpenCCU#1950] [🇩🇪 Fragen&Antworten] [answered]
Hallo zusammen,

beim Design meines Smarthomes habe ich mich bewusst gegen eine Lösung eines einzelnen Herstellers entschieden, da kein Hersteller für alles die passenden Geräte oder Konfigurationsmöglichkeiten besitzt. Stattdessen setze ich auf eine Reihe verschiedener Systeme, welche über HomeAssistant verknüpft sind. Neben Shellys, Phillips Hue Leuchten habe ich etwa auch Aqara Fenstersensoren und Thermometer sowie HmIP-Thermostate im Einsatz. Alles läuft auf einem RaspberryPi, zusammen mit 

Grundsätzlich funktioniert hier so gut wie alles wie vorgesehen, indem alles durch HomeAssistant koordiniert wird.
Eine Funktion scheint in diesem Setup aber nicht möglich zu sein: Die Möglichkeit, ein anderes Thermometer als Ist-Zustand für die HmIP-Thermostate zu verwenden. Das Thermostat erlaubt dies scheinbar ausschließlich als Direktverknüpfung mit einem HmIP-Wandthermostat. Diese möchte ich mir sowohl aus Kosten- als auch aus Designgründen aber nur sehr ungern anschaffen. Zumal ich die Aqara-Thermometer ja bereits habe und diese tadellos ihren Dienst verrichten.

Ich würde daher gerne Ideen sammeln, welche Lösungsmöglichkeiten es hier gibt.
So bin ich zB auf CUxD gestoßen, über welches ein virtuelles Gerät erzeugt werden kann. Leider können hierüber keine HmIP-Geräte erzeugt werden, sodass der Ansatz nicht ans Ziel führt.

Ebenfalls interessant fand ich die virtuellen Taster in der CCU. Diese gibt es auch in einer HmIP-Variante und lassen sich als Direktverknüpfung verknüpfen. Leider gibt es aber eben ausschließlich Taster, keine anderen Gerätetypen.

Ich freue mich auf eure Ratschläge :)

**Comment by SukramJ** [comment]
Für die Geizvariante gibt es vielleicht [hier](https://community.home-assistant.io/t/homematic-ip-local-reduce-room-temperature-when-a-window-or-door-is-opened/373305) bzw. [hier](https://community.home-assistant.io/t/homematic-reduce-room-temperature-when-a-window-or-door-is-opened/341053) Anregungen.

Hier geht es allerdings nur um den WINDOW_STATE und nicht um die ACTUAL_TEMPERATURE.
Die ACTUAL_TEMPERATURE ist ein nur-lese Attribute das API seitig nicht geändert werden kann.

**Comment by TheBoringGuy1908** [comment]
Danke für den Vorschlag!

Den WINDOW_STATE habe ich so bereits erfolgreich setzen können. Mir geht es aber tatsächlich um die ACTUAL_TEMPERATURE. Da ich diese nicht direkt setzen kann, hatte ich gehofft, dass es vielleicht eine Möglichkeit gibt, dem Thermostat ein Wandthermometer als Verknüpfungspartner "vorzugaukeln", so ähnlich wie die CCU ja einige Tasten vorgaugelt, die es auch nicht wirklich gibt.

**Reply by Baxxy13** [reply]
Es kann hier keine Ratschläge geben. 
Du hast es doch selbst schon geschrieben.
Ein IP-Heizkörperthermostat nutzt entweder seinen eigenen IST-Wert oder eben den eines verknüpften IP-Wandthermostat's.
Was anderes geht nicht. 
(Bei HM konnte man auch einen einfachen HM-Temperatursensor als ISTwertgeber verknüpfen.

**Reply by TheBoringGuy1908** [reply]
Mir ist klar, dass das innerhalb der "regulären" Homematic-Welt nicht geht.

Mir geht es hier aber unter anderem auch um die Diskussion, welche *technischen* Gründe dagegen sprechen, dass zB die CCU als Partner für eine Direktverknüpfung herhält. Damit meine ich jetzt nicht "weil es halt die CCU und kein Sensor ist". Anhand der virtuellen Taster sieht man ja, dass es rein theoretisch ja schon irgendwie geht.
Spannend ist hier also die Frage, wie die virtuellen Taster implementiert wurden, und ob man nicht auf ähnliche Art und Weise weitere Kanäle erzeugen könnte, welche andere Daten, wie eben die eines Thermometers, bereithalten.

Eine Lösung muss ja auch nicht jetzt sofort existieren. Vielmehr möchte ich eher gemeinsam mit der Community überlegen, wie für die Zukunft (wenn auch erstmal nur rein theoretisch) eine passende Lösung aussehen könnte.

**Reply by jp112sdl** [reply]
> Anhand der virtuellen Taster sieht man ja, dass es rein theoretisch ja schon irgendwie geht.

Das ist so vom Hersteller closed source so implementiert. VIRTUAL_KEY ist das einzige, was die CCU selbst als Peer senden und wofür die CCU als Peer dienen kann.

**Reply by holzwurm83** [reply]
Hallo,

seid ihr hierzu weiter gekommen? Ich komme aus der Fhem Welt und da war es in der tat möglich die Viertuellen Kanäle die fhem bereit gestellt hat mit den Kanälen eines Thermostats zu verknüpfen. Hatte z.B. die Fensterkanäle direkt verknüpft da die alte Homematic Version keine Datenpunkte besitzt um diese anzusteuern.

**Reply by opl-git** [reply]
Hallo zusammen,
ich bin auch interessiert, ob es hier etwas neues gibt.
Komme auch von FHEM und hier war es möglich virtuelle Sensoren anzulegen und mit dem Thermostat zu verknüpfen.
https://wiki.fhem.de/wiki/HM-CC-RT-DN_Funk-Heizk%C3%B6rperthermostat

**Reply by Baxxy13** [reply]
Es gibt nichts neues, und wird es vermutlich auch nie.
Die HM / HmIP - Geräte wollen halt unter sich bleiben. 😉

**Comment by opl-git** [comment]
Schade.
Komisch, dass es unter FHEM problemlos funktioniert.

**Reply by jp112sdl** [reply]
Was genau funktioniert unter FHEM problemlos?
Auch FHEM schafft es nicht, dem HMIPServer homematice-fremde Geräte unterzujubeln.

**Comment by opl-git** [comment]
Mit meinen HM-CC-RT-DN funktioniert es unter FHEM mit virtuellen Sensoren ziemlich gut.
Leider habe ich mit den IP Geräten keine Erfahrungen.

**Reply by jens-maus** [reply, marked as answer]
> Leider habe ich mit den IP Geräten keine Erfahrungen.

Aber genau darum geht es ja (Siehe Titel des Themas): Eben "eigene" HmIP Sensoren/Geräte einzubinden bzw. zu entwickeln. Und das geht nicht und wird es vmtl. auch nie geben. Ergo: Thema erledigt.

**Reply by opl-git** [reply]
Danke für die Klärung!
